pydatastructs/trees/heaps.py

```diff
@@ -83,6 +84,9 @@ def __new__(cls, elements=None, heap_property="min", d=4):
             raise ValueError("%s is invalid heap property"%(heap_property))
         if elements is None:
             elements = DynamicOneDimensionalArray(TreeNode, 0)
+        else:
+            if not all(map(lambda x: _check_type(x, TreeNode), elements)):
+                raise ValueError("Expect a list/tuple of TreeNode got %s"%(elements))

```

Comment:
Master should be merged as these changes are already there.

Reply:
thats a mess, I will remove them